tools/scripts/lock-api-state.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -ex
+
+# Buf image binary
+BUF_IMAGE=buf.image.bin
+PROJECT_ROOT=$(pwd)
+
+## lock in current protobuf state
+cd proto
+make gen-buf-image
+git add $BUF_IMAGE
+git commit -m "chore: lock api state for backward compatibility check" || echo "buf image is already up to date"
+cd $PROJECT_ROOT
```

Comment:
this line is actually useless unless you `source` the file, which should not be a concern.  Though if you did need to do this, it's generally better to use `pushd` and `popd`, so you don't have to store paths unnecessarily.

Reply:
yeah it was needed before since we were calling bumpversion but no longer needed. if we do not change the working directory we'd also need to prefix the BUF_IMAGE path.